kong/runloop/plugin_servers/process.lua

```diff
@@ -237,6 +238,34 @@ function proc_mgmt.pluginserver_timer(premature, server_def)
   kong.log.notice("Exiting: pluginserver '", server_def.name, "' not respawned.")
 end
 
+function proc_mgmt.connection_check_timer(premature, server_def)
+  if premature then
+    return
+  end
+
+  if ngx.config.subsystem ~= "http" then
+    return
+  end
+
+  local timeout = 500
+  local step = 0.1
+
+  local time = 0
+  local uri = "unix:" .. server_def.socket
+  local c, err
+  while time < timeout do
+    c, err = ngx.socket.connect(uri)
+    ngx.log(ngx.ERR, err)
```

Comment:
If socket connect successfully, we should not log any message. Should we check `err` then log it?